src/datasets/arrow_dataset.py

```diff
@@ -3242,6 +3242,116 @@ def shuffle(
+            Population per class.
+        n_draws : int
+            Number of draws (samples to draw) from the overall population.
+        rng : random state
+            Used to break ties.
+        Returns
+        -------
+        sampled_classes : ndarray of int
+            Number of samples drawn from each class.
+            np.sum(sampled_classes) == n_draws
+
+        """
+        # this computes a bad approximation to the mode of the
+        # multivariate hypergeometric given by class_counts and n_draws
+        continuous = n_draws * class_counts / class_counts.sum()
+        # floored means we don't overshoot n_samples, but probably undershoot
+        floored = np.floor(continuous)
+        # we add samples according to how much "left over" probability
+        # they had, until we arrive at n_samples
+        need_to_add = int(n_draws - floored.sum())
+        if need_to_add > 0:
+            remainder = continuous - floored
+            values = np.sort(np.unique(remainder))[::-1]
+            # add according to remainder, but break ties
+            # randomly to avoid biases
+            for value in values:
+                (inds,) = np.where(remainder == value)
+                # if we need_to_add less than what's in inds
+                # we draw randomly from them.
+                # if we need to add more, we add them all and
+                # go to the next value
+                add_now = min(len(inds), need_to_add)
+                inds = rng.choice(inds, size=add_now, replace=False)
+                floored[inds] += 1
+                need_to_add -= add_now
+                if need_to_add == 0:
+                    break
+        return floored.astype(np.int)
+
+    def _stratified_shuffle_split_generate_indices(self, y, n_train, n_test, rng, n_splits=10):
+        """
+
+        Provides train/test indices to split data in train/test sets.
+        It's reference is taken from StratifiedShuffleSplit implementation
+        of scikit-learn library.
+
+        Args
+        ----------
+
+        n_train : int,
+            represents the absolute number of train samples.
+
+        n_test : int,
+            represents the absolute number of test samples.
+
+        random_state : int or RandomState instance, default=None
+            Controls the randomness of the training and testing indices produced.
+            Pass an int for reproducible output across multiple function calls.
+
+        n_splits : int, default=10
+            Number of re-shuffling & splitting iterations.
+        """
+        classes, y_indices = np.unique(y, return_inverse=True)
+        n_classes = classes.shape[0]
+        class_counts = np.bincount(y_indices)
+        if np.min(class_counts) < 2:
+            raise ValueError(
+                "The least populated class in y has only 1"
+                " member, which is too few. The minimum"
+                " number of groups for any class cannot"
+                " be less than 2."
```

Comment:
Let's improve this error message by specifying the actual `stratify` column name instead of `y`

Reply:
Fixed